R/checkmate.R

```diff
@@ -21,6 +21,107 @@
+#' Elements of `...` must be numeric vectors or scalars.
+#'
+#' This function performs an element-by-element comparison of the first object
+#' provided in `...` with every other object in `...` and returns `TRUE` if all
+#' comparisons are equal within a given tolerance and `FALSE` otherwise.
+#'
+#' @param ... (`numeric`)\cr vectors to be compared
+#' @param tol (`numeric`)\cr the maximum difference to be tolerated when
+#' judging equality
+#'
+#' @note If there are any missing or infinite values in `...`, this function
+#'   returns `FALSE`, regardless of the values of other elements in `...`.
+#'
+#' @note If elements in `...` are not all of the same length, `FALSE` is returned.
+#'
+#' @return TRUE if all element-by-element differences are less than `tolerance`
+#' in magnitude, `FALSE` otherwise.
+#' @seealso [`assertions`] for more details.
+#'
+#' @export
+#' @examples
+#' check_equal(1:2, 1:2) # TRUE
+#' check_equal(1:2, 2:3) # "Not all equal"
+#' check_equal(Inf, Inf) # "Not all equal"
+#' check_equal(0.01, 0.02) # "Not all equal"
+#' check_equal(0.01, 0.02, tol = 0.05) # TRUE
+#' check_equal(1, c(1, 1)) # "Not all equal"
+check_equal <- function(..., tol = sqrt(.Machine$double.eps)) {
+  dot_args <- list(...)
+
+  sapply(dot_args, assert_numeric)
+
+  tmp <- sapply(dot_args, length)
+  if (min(tmp) != max(tmp)) {
+    return("Not all of same length")
+  }
+  if (any(sapply(dot_args, is.na))) {
+    return("Some entries NA")
+  }
+  if (any(sapply(dot_args, is.infinite))) {
+    return("Not all entries finite")
+  }
+  if (!all(sapply(dot_args, test_numeric))) {
+    return("Not all numeric")
+  }
+
+  rv <- test_true(
+    all(
+      sapply(
+        2:length(dot_args),
+        function(z) abs(dot_args[[1]] - dot_args[[z]]) < tol
+      )
+    )
+  )
+  if (rv) {
+    return(TRUE)
+  } else {
+    return("Not all equal")
+  }
+}
+
+#  assert equality
+
+#' Assert That All Arguments Are Equal
+#'
+#' @description `r lifecycle::badge("experimental")`
+#' Elements of `...` must be numeric vectors or scalars.
+#'
+#' This function performs an element-by-element comparison of the first object
+#' provided in `...` with every other object in `...` and throws an error if they
+#' are not.
+#'
+#' @param ... (`numeric`)\cr vectors to be compared
+#' @param tol (`numeric`)\cr the maximum difference to be tolerated when
+#' judging equality
+#'
+#' @note If there are any missing or infinite values in `...`, this function
+#'   throws an error, regardless of the values of other elements in `...`.
+#'
+#' @note If elements in `...` are not all of the same length, an error is thrown.
+#'
+#' @return `list(...)`, invisibly.
+#' @seealso [`assertions`] for more details.
+#' @inheritParams checkmate::assert_numeric
+#'
+#' @export
+#' @examples
+#' assert_equal(1:2, 1:2) # no error
+#' assert_equal(0.01, 0.02, tol = 0.05) # no error
+# nolint start
+assert_equal <- function(..., tol = sqrt(.Machine$double.eps), .var.name = vname(x), add = NULL) {
```

Comment:
better use `makeAssertionFunction` or add comment why that would not work

Reply:
> better use `makeAssertionFunction` or add comment why that would not work

`assert_equal <- makeAssertionFunction(check_equal)` fails with error "Error in `checkmate::makeAssertion(..., res, .var.name, add)`: unused argument (add)".  I *think* this is because of the use of `...` in `check_equal`.